1.2/percentile_test.go

```diff
@@ -0,0 +1,59 @@
+package percentile
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestGeneratePercentile(t *testing.T) {
+	var tests = []struct {
+		fn percentile
+		p  uint
+	}{
+		{generatePercentile(.05), 15},
+		{generatePercentile(.3), 20},
+		{generatePercentile(.4), 20},
+		{generatePercentile(.5), 35},
+		{generatePercentile(1.0), 50},
+	}
+	data := []uint{15, 20, 35, 40, 50}
```

Comment:
Does your function work correctly if these are not ordered?

Reply:
This is assuming the input is ordered, but I can add a sort method